### PR TITLE
expose updateTime on runs, default instantiate GrapheneRun with records

### PR DESCRIPTION
**Pull request by prha**
## Summary
In order to display the `RunTime` component for a given run without accessing `RunStatsSnapshot`, we need to approximate the queued time / launch time by grabbing updateTime for runs with queued / starting status.

## Test Plan
BK

### CODE REVIEW

File: python_modules/dagster-graphql/dagster_graphql/implementation/fetch_runs.py [hunk 1]
```diff
@@ -87,24 +87,27 @@ def get_runs(graphene_info, filters, cursor=None, limit=None):
...
-    runs = []
 
     if filters and filters.run_ids and len(filters.run_ids) == 1:
         run = instance.get_run_by_id(filters.run_ids[0])
         if run:
-            runs = [run]
-    elif filters and (
+            return [GrapheneRun(run)]
+        else:
+            return []
```
Comment (alangenfeld) [inline python_modules/dagster-graphql/dagster_graphql/implementation/fetch_runs.py:96]: should we still do this "shortcut" path? seems bad now that we want to root on `record`

File: python_modules/dagster-graphql/dagster_graphql/schema/pipelines/pipeline.py [hunk 2]
```diff
@@ -229,6 +230,7 @@ class GrapheneRun(graphene.ObjectType):
     )
     startTime = graphene.Float()
     endTime = graphene.Float()
+    updateTime = graphene.Float()
 
     class Meta:
         interfaces = (GraphenePipelineRun,)
```
Comment (alangenfeld) [inline python_modules/dagster-graphql/dagster_graphql/schema/pipelines/pipeline.py:236]: <i class='graphite__hidden'>[Re: line 240]<br/><br/></i>is that callsite above the only reason we need this?

File: python_modules/dagster-graphql/dagster_graphql/schema/pipelines/pipeline.py [hunk 3]
```diff
@@ -347,7 +349,9 @@ def resolve_events(self, graphene_info, after=-1):
 
     def _get_run_record(self, instance):
         if not self._run_record:
-            self._run_record = instance.get_run_records(run_ids=[self.runId])[0]
+            self._run_record = instance.get_run_records(PipelineRunsFilter(run_ids=[self.run_id]))[
+                0
+            ]
```
Comment (alangenfeld) [inline python_modules/dagster-graphql/dagster_graphql/schema/pipelines/pipeline.py:354]: related

**Review by alangenfeld — CHANGES_REQUESTED** [review]
How much work would it take to always be backed by run_record?

File: python_modules/dagster-graphql/dagster_graphql/schema/pipelines/pipeline.py (hunk 2 above)
Comment (prha) [reply python_modules/dagster-graphql/dagster_graphql/schema/pipelines/pipeline.py:236]: Yeah, updateTime is just a hack to get the `RunTime` component working without fetching stats

File: python_modules/dagster-graphql/dagster_graphql/implementation/fetch_runs.py (hunk 1 above)
Comment (prha) [reply python_modules/dagster-graphql/dagster_graphql/implementation/fetch_runs.py:96]: yeahhhhhhh.... I should do this, I was just being lazy.

File: python_modules/dagster-graphql/dagster_graphql/schema/pipelines/pipeline.py [hunk 4]
```diff
@@ -229,20 +225,21 @@ class GrapheneRun(graphene.ObjectType):
...
+    updateTime = graphene.Float()
 
     class Meta:
         interfaces = (GraphenePipelineRun,)
         name = "Run"
 
-    def __init__(self, run):
-        pipeline_run = run.pipeline_run if isinstance(run, RunRecord) else run
+    def __init__(self, record: RunRecord):
+        pipeline_run = record.pipeline_run
```
Comment (alangenfeld) [inline python_modules/dagster-graphql/dagster_graphql/schema/pipelines/pipeline.py:235]: maybe `check` call here since this package isn't mypy-ed in bk

**Review by alangenfeld — APPROVED** [review]
confident in the existing test suite?

**Pull request merged**